### PR TITLE
integration, vendor: prepare for use of e2e-harness pkg/release

### PR DESCRIPTION
**Pull request by kopiczko**

### CODE REVIEW

File: integration/setup/config.go [hunk 1]
```diff
@@ -1,16 +1,174 @@
...
-	AWSClient *awsclient.Client
+	AWSClient *e2eclientsaws.Client
 	Guest     *framework.Guest
 	Host      *framework.Host
+	Release   *release.Release
 	Resource  *resource.Resource
 	Logger    micrologger.Logger
 }
+
+func NewConfig() (Config, error) {
```
Comment (kopiczko) [inline integration/setup/config.go]: I'd like to initialize & fix stuff once. This comes in handy.

File: integration/setup/config.go [hunk 2]
```diff
@@ -1,16 +1,174 @@
...
+			return Config{}, microerror.Mask(err)
+		}
+
+	}
+
+	c := Config{
+		AWSClient: awsClient,
+		Guest:     guest,
+		Host:      host,
+		Release:   newRelease,
```
Comment (kopiczko) [inline integration/setup/config.go]: Here we have `Release` object ready to be used.

File: integration/setup/config.go (hunk 1 above)
Comment (MarcelMue) [reply integration/setup/config.go]: I think this should be in the `e2e-harness` itself as kind of a framework. Everything except the `AWSClient` is going to be necessary in any e2e test. (The `Guest` stuff should be able to be empty in e2e-harness though).

File: integration/setup/config.go [hunk 3]
```diff
@@ -1,16 +1,174 @@
...
+	{
+		c := micrologger.Config{}
+
+		logger, err = micrologger.New(c)
+		if err != nil {
+			return Config{}, microerror.Mask(err)
+		}
+	}
+
+	var apprClient *apprclient.Client
```
Comment (MarcelMue) [inline integration/setup/config.go]: This is actually defaulted in `Release`, but still can't hurt to do it explicitly here.

File: integration/setup/config.go [hunk 4]
```diff
@@ -1,16 +1,174 @@
...
+		newRelease, err = release.New(c)
+		if err != nil {
+			return Config{}, microerror.Mask(err)
+		}
+	}
+
+	var newResource *resource.Resource
+	{
+		c := resource.Config{
+
```
Comment (xh3b4sd) [inline integration/setup/config.go]: Lets drop this.

File: integration/setup/config.go [hunk 5]
```diff
@@ -1,16 +1,174 @@
...
+	}
+
+	var fileLogger *filelogger.FileLogger
+	{
+		fc := filelogger.Config{
+			Backoff:   backoff.NewExponential(backoff.ShortMaxWait, backoff.LongMaxInterval),
+			K8sClient: host.K8sClient(),
+			Logger:    logger,
+		}
+		fileLogger, err = filelogger.New(fc)
```
Comment (xh3b4sd) [inline integration/setup/config.go]: Here I would like to keep the extra lines. Below the same.

File: integration/test/draining/draining_test.go [hunk 6]
```diff
@@ -231,7 +231,7 @@ func Test_Draining(t *testing.T) {
 }
 
 func numberOfMasters(clusterID string) (int, error) {
-	cluster, err := h.AWSCluster(clusterID)
+	cluster, err := config.Host.AWSCluster(clusterID)
```
Comment (xh3b4sd) [inline integration/test/draining/draining_test.go]: Just to keep this in mind. The wording is still off here. Should probably be `Tenant` and `ControlPlane`.

File: integration/setup/config.go (hunk 3 above)
Comment (kopiczko) [reply integration/setup/config.go]: It hurts me :D I will remove.

**Comment by kopiczko** [comment]
- Previous workflow was fine. Only scaling test didn't work.
- Scaling test passed on my local setup.
- We tried to debug failed cluster with @r7vme , but it was deleted before we were able to figure anything out. I couldn't SSH. We checked my pub key is in the CR in circle and I could SSH to the node using the same method in other cluster.
- Maybe we should move SSH users creation to the "small cloud config".  That looks like "big cc" fails for w/e reason and we can't SSH to debug that.
- I can't recreate the thing on my local setup where I have KEEP_RESOURCES=true setting.

So I just merge to keep going. Let's have other quick wins first.

**Comment by kopiczko** [comment]
BTW this is very similar observation to what we have in cluster-operator.

**Pull request merged**